patches/blockprotocol+0.0.7.patch

```diff
@@ -67,3 +74,163 @@ index 20a9bbd..eb3a4e4 100644
+   linkedAggregations?: BlockProtocolLinkedAggregation[];
+   linkedEntities?: BlockProtocolEntity[];
+   linkGroups?: BlockProtocolLinkGroup[];
+-} & BlockProtocolFunctions;
++} & BlockProtocolFunctions & BlockProtocolStyling;
+diff --git a/node_modules/blockprotocol/styling.d.ts b/node_modules/blockprotocol/styling.d.ts
+new file mode 100644
+index 0000000..e7f656f
+--- /dev/null
++++ b/node_modules/blockprotocol/styling.d.ts
+@@ -0,0 +1,140 @@
++/**
++ * Block Protocol theme is inspired by MUI
++ * @see https://mui.com/customization/theming/
++ * @see 
++ */
++
++export interface Color {
++  50: string;
++  100: string;
++  200: string;
++  300: string;
++  400: string;
++  500: string;
++  600: string;
++  700: string;
++  800: string;
++  900: string;
++  A100: string;
++  A200: string;
++  A400: string;
++  A700: string;
++}
++
++export type ColorPartial = Partial<Color>;
++
++export interface SimplePaletteColorOptions {
++  light?: string;
++  main: string;
++  dark?: string;
++  contrastText?: string;
++}
++
++export type PaletteMode = 'light' | 'dark';
++
++export type PaletteTonalOffset =
++  | number
++  | {
++      light: number;
++      dark: number;
++    };
++
++export type PaletteColorOptions = SimplePaletteColorOptions | ColorPartial;
++
++export interface CommonColors {
++  black: string;
++  white: string;
++}
++
++export interface TypeAction {
++  active: string;
++  hover: string;
++  hoverOpacity: number;
++  selected: string;
++  selectedOpacity: number;
++  disabled: string;
++  disabledOpacity: number;
++  disabledBackground: string;
++  focus: string;
++  focusOpacity: number;
++  activatedOpacity: number;
++}
++
++export interface TypeBackground {
++  default: string;
++  paper: string;
++}
++
++export interface TypeText {
++  primary: string;
++  secondary: string;
++  disabled: string;
++}
++
++export type PaletteOptions = {
++  primary?: PaletteColorOptions;
++  secondary?: PaletteColorOptions;
++  error?: PaletteColorOptions;
++  warning?: PaletteColorOptions;
++  info?: PaletteColorOptions;
++  success?: PaletteColorOptions;
++  mode?: PaletteMode;
++  tonalOffset?: PaletteTonalOffset;
++  contrastThreshold?: number;
++  common?: Partial<CommonColors>;
++  grey?: ColorPartial;
++  text?: Partial<TypeText>;
++  divider?: string;
++  action?: Partial<TypeAction>;
++  background?: Partial<TypeBackground>;
++}
++
++export type Shadows = [
++  'none',
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++  string,
++];
++
++export interface ZIndex {
++  mobileStepper: number;
++  speedDial: number;
++  appBar: number;
++  drawer: number;
++  modal: number;
++  snackbar: number;
++  tooltip: number;
++}
++
++export type ZIndexOptions = Partial<ZIndex>;
```

Comment:
do we want blocks knowing about these? if so, why? if anything I'd assume that maybe we want to give blocks a `maxZIndex` so that they don't render things above stuff the app wants to be on top

We could leave them out at first until we identify a need for them.

Reply:
I have copy-pasted `ZIndex` from MUI to reuse object shapes as close as possible, thus compatible with each other.

Not sure if `maxZIndex` would be needed assuming blocks are inside iframes.

Reply:
That's fine for now, but my assumption here is that the end product should be something which looks like a generalized solution rather than "the same as MUI" (in some places the same will be fine, in others - like here - it looks odd)